### DISCUSSION TITLE
How to control Schneider Electric / Wiser  CCTFR6400 Thermostat?

### DISCUSSION THREAD

**Discussion by wsbtak** [arendst/Tasmota#21848] [Q&A] [answered]
I have a new Wiser thermostat that I would like to operate with Z2T. Pairing works and I receive values (temperature, humidity, and battery status), but I cannot set any values (screen lock, desired temperature, on-demand heating or heat icon). Additionally, a red triangle appears and the thermostat blinks red sporadically, which indicates incomplete pairing.

Do I need to create my own ZB definition file? If so, how should I proceed and what needs to be included? What should the ZBSend command look like to set the desired temperature, etc.?

References:
https://zigbee.blakadder.com/Schneider_Electric_CCTFR6400.html
https://github.com/Koenkk/zigbee2mqtt/issues/11340

> 10:33:51.840 RSL: SENSOR = {"ZbReceived":{"0x019A":{"Device":"0x019A","Cluster":513,"Read":[18,28,27,8],"ReadNames":{"OccupiedHeatingSetpoint":true,"SystemMode":true,"ControlSequenceOfOperation":true,"PIHeatingDemand":true},"Endpoint":1,"LinkQuality":147}}}
10:33:51.890 RSL: SENSOR = {"ZbReceived":{"0x019A":{"Device":"0x019A","Cluster":513,"Read":[57616],"ReadNames":{},"Endpoint":1,"LinkQuality":147}}}
10:33:52.344 RSL: SENSOR = {"ZbReceived":{"0x019A":{"Device":"0x019A","Humidity":66.16,"Endpoint":1,"LinkQuality":147}}}
10:33:56.936 RSL: SENSOR = {"ZbReceived":{"0x019A":{"Device":"0x019A","FE03/0020":"UI,ScreenSleep","Endpoint":1,"LinkQuality":147}}}
10:34:23.641 RSL: SENSOR = {"ZbReceived":{"0x019A":{"Device":"0x019A","Cluster":0,"Read":[0],"ReadNames":{"ZCLVersion":true},"Endpoint":1,"LinkQuality":147}}}
10:34:27.337 RSL: SENSOR = {"ZbReceived":{"0x019A":{"Device":"0x019A","FE03/0020":"ENV,-32768,2690,6396","LastMessageLQI":null,"LastMessageRSSI":-25,"0B05/E001":0,"Humidity":63.96,"Endpoint":1,"LinkQuality":147}}}

> 10:58:01.492 CMD: zbinfo 0x019A
10:58:01.497 RSL: SENSOR = {"ZbInfo":{"0x019A":{"Device":"0x019A","IEEEAddr":"0x8C6FB9FFFE2B8DA4","ModelId":"Thermostat","Manufacturer":"Schneider Electric","Endpoints":[1],"Config":["T01"],"Temperature":26.87,"Humidity":62.9,"Reachable":true,"BatteryPercentage":50,"BatteryLastSeenEpoch":1721984772,"LastSeen":22,"LastSeenEpoch":1721987859,"LinkQuality":136}}}

**Comment by wsbtak** [comment]
UPDATE: Figured out how to set screen lock:
`ZbSend {"Device":"0x019A","Write":{"ThermostatKeypadLockout":1}}`

However OccupiedHeatingSetpoint and PIHeatingDemand are both resulting in "UNSUPPORTED_ATTRIBUTE"
```
14:52:22.349 CMD: ZbSend {"Device":"0x019A","Write":{"OccupiedHeatingSetpoint":21}}
14:52:22.353 RSL: RESULT = {"ZbSend":"Done"}
14:52:22.549 RSL: RESULT = {"ZbResponse":{"Device":"0x019A","Command":"0201_04","Status":134,"StatusMessage":"UNSUPPORTED_ATTRIBUTE","Endpoint":1,"LinkQuality":147}}
```

**Reply by dhWasabi** [reply, marked as answer]
It has been determined that for the thermostat to be operational, it must be paired with either a TRV or a relay module from Wiser, specifically when using the original Wiser Zigbee Hub V2.
